src/query/service/src/interpreters/interpreter_table_create.rs

```diff
@@ -82,20 +82,21 @@ impl Interpreter for CreateTableInterpreter {
         let quota = quota_api.get_quota(MatchSeq::GE(0)).await?.data;
         let engine = self.plan.engine;
         let catalog = self.ctx.get_catalog(self.plan.catalog.as_str())?;
-        let tables = catalog
-            .list_tables(&self.plan.tenant, &self.plan.database)
-            .await?;
-        if quota.max_tables_per_database != 0
-            && tables.len() >= quota.max_tables_per_database as usize
-        {
-            return Err(ErrorCode::TenantQuotaExceeded(format!(
-                "Max tables per database quota exceeded: {}",
-                quota.max_tables_per_database
-            )));
-        };
-        let name_not_duplicate = tables
-            .iter()
-            .all(|table| table.name() != self.plan.table.as_str());
+        if quota.max_tables_per_database != 0 {
```

Comment:
```suggestion
        if quota.max_tables_per_database > 0 {
```

Reply:
I see this type is u32. It will never be neg?

```rust
// The max tables per database can be created in the tenant.
    pub max_tables_per_database: u32,
```

Reply:
Check greater than make the codes is more readable.